### PR TITLE
feat(go.mod): update badger dependencies

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- Currently, our project uses `badger/v2` and `ipfs/go-ds-badger2` and those packages are quite out of date and new version of them brings substantial performance improvements.

- Also we should update our `ChainSafe/chaindb` dependency once PR https://github.com/ChainSafe/chaindb/pull/24 was merged there then we will be fully in up to date version of badger

- The difference between this PR and Quentin's PR https://github.com/ChainSafe/gossamer/pull/3005 is that this one is a simple dependency update without much refactors

## Tests

N/A

## Issues

N/A

## Primary Reviewer

@timwu20

### CODE REVIEW

**Review by jimjbrettj — APPROVED** [review]
lgtm

File: dot/sync/worker.go [hunk 1]
```diff
@@ -26,11 +26,13 @@ type workerState struct {
...
 func newWorkerState() *workerState {
 	ctx, cancel := context.WithCancel(context.Background())
-	return &workerState{
+	ws := &workerState{
 		ctx:     ctx,
 		cancel:  cancel,
 		workers: make(map[uint64]*worker),
 	}
+
+	return ws
```
Comment (timwu20) [inline dot/sync/worker.go:35]: nit: unnecessary diff

**Review by timwu20 — COMMENTED** [review]
shouldn't this code import the changes you made to chaindb to also use badger v3?

**Comment by EclesioMeloJunior** [comment]
> shouldn't this code import the changes you made to chaindb to also use badger v3?

Yes, but I need to merge this PR https://github.com/ChainSafe/chaindb/pull/24 first before updating the dependency here

**Review by qdm12 — COMMENTED** [review]
I'm not sure why this was done/decided to be worked on, but it does feel like duplicate work (even though the PRs I opened are more refactors/fixes than just dep upgrade, the dependency upgrade on 2 repos + staging testing are duplicate work). It would had been maybe better to pick up #3088 and fix integration tests failing?

I'm not too opposed to this PR either since it might help the situation and is simpler to get merged though.

**Comment by EclesioMeloJunior** [comment]
@qdm12 agree with you, I came up with this dep upgrade once I identified that we were using an quite old badger version in both repositories, and I notice your PR https://github.com/ChainSafe/gossamer/pull/3005 lately, so I will close this PR in favor of https://github.com/ChainSafe/gossamer/pull/3088 and https://github.com/ChainSafe/gossamer/pull/3005 and then I will pick up https://github.com/ChainSafe/gossamer/pull/3088 and fix the tests on it, that sounds good? 

fyi @timwu20
